nmsg_message.pyx

```diff
@@ -281,10 +281,24 @@ cdef class message(object):
                 fields = [ self.fields[field_name] ]
 
             for i in range(0, len(fields)):
-                if type(fields[i]) == bytes:
-                    fields_enc = fields[i]
+                if isinstance(fields[i], bytes):
```

Comment:
I'm trying to persuade @kelvinatorr  that it would be cleaner to treat the encoding below, since it's type-driven--e.g. we wouldn't have to repeat this test for IP.  @hstern or @djw1149 any opinion?  Imagining a kind of "factory".

Reply:
I agree it is not a clean coding style, as it just a heuristic that the values are an IPv4 or IPv6 address and thus should not be utf-8 encoded.  There are other values that could parse as an IP address but are not really IP addresses.

Reply:
Ok; I changed it so it only encodes when the expected value is a string (or an enum with strings). The 2 new unittests still pass and examples/nmsg_message.py still works.

Reply:
Thanks for that change @kelvinatorr .